### PR TITLE
curl: load source assets in dev to avoid compiling

### PR DESCRIPTION
**Pull request by OliverJAsh**
This change means we can have a "refresh to reload" workflow using our current module loader, just like what we had with jspm. Still stuck on ES5 for now, but one thing at a time.

### CODE REVIEW

**Comment by sndrs** [comment]
magic :gem: :ghost: :+1:

File: Gruntfile.js [hunk 1]
```diff
@@ -99,7 +99,11 @@ module.exports = function (grunt) {
     grunt.registerTask('compile:js', function (fullCompile) {
         grunt.task.run(['clean:js', 'compile:inlineSvgs']);
 
-        grunt.task.run(['concurrent:requireJS', 'copy:javascript']);
+        if (!options.isDev) {
```
Comment (rich-nguyen) [inline Gruntfile.js]: i think if a dev is running this task, then it should do the `concurrent:requireJS` task. Why do we need the guard? is it for a watch?

Comment (sndrs) [inline Gruntfile.js]: it's just not used at all. we can still build them, but you'll never load them in dev

File: grunt-configs/copy.js [hunk 2]
```diff
@@ -31,11 +31,12 @@ module.exports = function (grunt, options) {
                         '**/*'
                     ],
                     dest: options.staticHashDir + 'javascripts/vendor'
-                },
-                {
+                },{
                     expand: true,
-                    cwd: options.requirejsDir,
-                    src: [
+                    cwd: options.isDev ? options.staticSrcDir + 'javascripts' : options.requirejsDir,
```
Comment (rich-nguyen) [inline grunt-configs/copy.js]: Why is a verbatim copy needed from `static/src` to `static/target`? Aren't we loading directly from `static/src`?

Comment (sndrs) [inline grunt-configs/copy.js]: `AssetMap` was expecting them in the map. this puts them in place so asset_hash includes them

File: Gruntfile.js (hunk 1 above)
Comment (rich-nguyen) [inline Gruntfile.js]: that's fine. it makes more sense to compile the js if you've run this task.

File: grunt-configs/copy.js (hunk 2 above)
Comment (rich-nguyen) [inline grunt-configs/copy.js]: We shouldn't be using `AssetMap`, because we don't need to resolve paths to hashed paths.

File: Gruntfile.js (hunk 1 above)
Comment (sndrs) [inline Gruntfile.js]: i'm not sure why? if you still compile the JS in `--dev`, you'll have to wait for it to compile but never download it. not sure what the benefit is? it was the same with JSPM, we never bundled in dev. same with uglifying etc?

Comment (sndrs) [inline Gruntfile.js]: hmmm, do you mean because you've run `compile:js`? i see your point...

**Comment by OliverJAsh** [comment]
Thanks for the changes @rich-nguyen and @sndrs :+1:

**Comment by sndrs** [comment]
:+1: from me. love this.

**Pull request merged**